code/backend/batch/function_app.py

```diff
@@ -10,7 +10,8 @@
 logging.captureWarnings(True)
 # Raising the azure log level to WARN as it is too verbose - https://github.com/Azure/azure-sdk-for-python/issues/9422
 logging.getLogger("azure").setLevel(os.environ.get("LOGLEVEL_AZURE", "WARN").upper())
-configure_azure_monitor()
+if os.getenv("APPLICATIONINSIGHTS_ENABLED", "false").lower() == "true":
```

Comment:
Can we use EnvHelper's APPLICATIONINSIGHTS_ENABLED field instead? Could save us from the .lower() and == "true" logic. @adamdougal do you see any issues with instantiating env helper in this file?

Reply:
Ah so we did that before. But then the `EnvHelper` gets initialised, with all of it's calls to KeyVault etc **before** application insights is configured, meaning we lose visibiliy on those calls :(

Reply:
@adamdougal - do we need visibility of those calls? Particularly if the projects switches to Key Vault references (see #625)

Reply:
I think it'd be good to keep it, and then revaluate after #625. We weren't really aware of the number of KeyVault calls we were making until we added this, and turned out it was quite a lot!

Reply:
I think another issue is that you are meant to call `configure_azure_monitor()` as early as possible, before importing any of the modules that you want to instrument, because the `configure_azure_monitor()` function essentially "monkeypatches" the modules to add instrumentation

Reply:
But it doesn't look like we're doing that currently anyway 🤷‍♂️

Reply:
Thanks for the discussion team. I have opted to keep it, and then revaluate after #625